build-system/tasks/cherry-pick.js

```diff
@@ -0,0 +1,179 @@
+/**
+ * Copyright 2020 The AMP HTML Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+'use strict';
+
+const argv = require('minimist')(process.argv.slice(2));
+const log = require('fancy-log');
+const {getOutput} = require('../common/exec');
+const {green, cyan, red, yellow} = require('ansi-colors');
+
+const RTV_PREFIXES = {
+  '00': 'experimental',
+  '01': 'stable',
+  '02': 'control',
+  '03': 'beta',
+  '04': 'nightly',
+  '05': 'nightly-control',
+};
+
+/**
+ * Executes a shell command, and logs an error message if the command fails.
+ *
+ * @param {string} cmd
+ * @param {string} msg
+ * @return {!Object}
+ */
+function execOrLog(cmd, msg) {
+  const result = getOutput(cmd);
+  if (result.status) {
+    log(yellow('ERROR:'), msg);
+    throw new Error(result.stderr);
+  }
+
+  return result;
+}
+
+/**
+ * Determines the name of the cherry-pick branch.
+ *
+ * @param {string} rtv
+ * @param {Array<string>} commits
+ * @return {string}
+ */
+function cherryPickBranchName(rtv, commits) {
+  const prefix = rtv.substr(0, 2);
+  const channel = RTV_PREFIXES[prefix] || 'unknown';
+  const [date] = new Date().toISOString().split('T');
+
+  return `amp-release-${date}-${channel}+${commits.length}`;
```

Comment:
What do you think about changing the branch format to `amp-release-${new-amp-version}` where new-amp-version is the given amp-version plus 1?

Reply:
I'm quite open to it. Would also support leaving the branch name, but adding the tag here.

Also, wouldn't it be the version + # cherry-picks? ie.
```
version = Number(rtv.substr(2)) + commits.length
return `amp-release-${version}`
```

Reply:
Hmm, I don't think it should increment by # of commits, but the # of git cherry-pick commands. Because if we have releases (2217) and (2217+2), I would assume that (2217+1) exists.

Reply:
I like `amp-release-${version}` more and more, it would make it easy for release automation to check if a cherry pick branch exists when it has to decide which branch to promote to opt-in.

It also simplifies this task because you won't need RTV_PREFIXES or channel names :)

Reply:
I do like the idea of not needing to guess at the prefix. As for what to increment by, the # of git cherry-picks will be the same as the # of commits (unless one fails and the task aborts). If there are three commits to cherry-pick, with three associated CP issues, the RTV should end in `003` even if there's no `002` or `001`. I think this should be fine? There's little benefit to release and tag `001` if we know that two more CPs are needed

Reply:
Updated this to use the `amp-release-${version}` form, PTAL

Reply:
Ahh, I think I understand our differences! Let's agree on these design details first - 

1. Each CP issue can have 1 or more commit. Most often we only cherry pick 1 commit. But we support 1 or more per issue.

2. gulp cherry-pick is called for every CP issue. We  should not combine multiple CP issues into 1 gulp cherry-pick.

3. The release number should increment by # of CP issues.

Reply:
Thanks for clarifying these pieces! I wasn't aware of 2; that is, I was under the impression all the CP commits would be added at once. Now it make sense why we'd just be incrementing the last 3 digits. Will update.

Reply:
Updated!